content/en/docs/Concepts/dependency-resolution/catalogsource-priority.md

```diff
@@ -0,0 +1,32 @@
+---
+title: "CatalogSource Priority"
+weight: 2
+---
+
+As part of the SAT dependency resolver, OLM can choose from a wide range of catalogs to resolve dependencies. CatalogSource is a CR that defines where the catalog comes from and, therefore, can infer the amount of confidence cluster administrators or operator subscribers would want to rely on certain catalogs. OLM would like to explicitly weight catalogs for the dependency resolver so that certain catalogs can be priorities. The default CatalogSources such as Red Hat, Certified, and community catalogs will be assigned default priorities and subject to changes. 
```

Comment:
> the SAT dependency resolver

The translation of dependency resolution problems into SAT is an implementation detail. Leaving that out may help to avoid confusing some readers.

> a wide range of catalogs

It's important to be specific about which catalogs are used for dependency resolution. If someone is reading this section of the documentation, they will want to know that dependencies are satisfied from any of the CatalogSources in the same namespace as a Subscription _plus_ any CatalogSources in the configurable global namespace.

Side note: The default global namespace should be more upstream-friendly. It's currently "openshift-operator-lifecycle-manager."

> OLM would like to explicitly weight catalogs

I wouldn't say that OLM is weighting catalogs -- it's honoring explicit priorities set by users. Also, I think it would be helpful in general to avoid using "weight" interchangeably with "priority," and to instead use "priority" everywhere that catalog preferences are discussed.

> default CatalogSources such as Red Hat, Certified, and community catalogs

This, and the other mention of these specific catalogs, belongs in the OpenShift docs, not here.

Reply:
yeah, but the so-called openshift doc lives on the google dive, somehow our code is opensource for those parts as well. we might want to redefine a new place for some of those docs